Classes/Plugins/Coins/Coins.php

```diff
@@ -0,0 +1,250 @@
+ * For the full copyright and license information, please read the
+ * LICENSE.txt file that was distributed with this source code.
+ *
+ * The TYPO3 project - inspiring people to share!
+ */
+
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+
+/**
+ * Plugin 'DPF: Coins' for the 'dlf / dpf' extension.
+ *
+ * @author    Erik Sommer <erik.sommer@slub-dresden.de>
+ * @package    TYPO3
+ * @subpackage    tx_dpf
+ * @access    public
+ */
+class Coins extends \Kitodo\Dlf\Common\AbstractPlugin
+{
+    public $scriptRelPath = 'Classes/Plugins/Coins.php';
+
+    /**
+     * The main method of the PlugIn
+     *
+     * @access    public
+     *
+     * @param    string        $content: The PlugIn content
+     * @param    array        $conf: The PlugIn configuration
+     *
+     * @return    string        The content that is displayed on the website
+     */
+    public function main($content, $conf)
+    {
+        $this->init($conf);
+
+        // get the tx_dpf.settings too
+        // Flexform wins over TS
+        $dpfTSconfig = $GLOBALS['TSFE']->tmpl->setup['plugin.']['tx_dpf.'];
+        if (is_array($dpfTSconfig['settings.'])) {
+            \TYPO3\CMS\Core\Utility\ArrayUtility::mergeRecursiveWithOverrule($dpfTSconfig['settings.'], $this->conf, true, false);
+            $this->conf = $dpfTSconfig['settings.'];
+        }
+
+        $this->setCache(true);
+
+        $this->loadDocument();
+        if ($this->doc === null) {
+            return;
+        } else {
+            // Set default values if not set.
+            if (!isset($this->conf['rootline'])) {
+                $this->conf['rootline'] = 0;
+            }
+        }
+
+        $metadata = array();
+        $metadata = $this->doc->getTitleData($this->conf['pages']);
+
+        $metadata['_id'] = $this->doc->toplevelId;
+        if (empty($metadata)) {
+            if (TYPO3_DLOG) {
+                GeneralUtility::devLog(
+                    '[tx_dpf_metatags->main(' . $content . ', [data])] No metadata found for document with UID "' . $this->doc->uid . '"',
+                    'tx_dpf',
+                    SYSLOG_SEVERITY_WARNING,
+                    $conf
+                );
+            }
+            return;
+        }
+
+        return $this->generateCoins($metadata);
+    }
+
+    /**
+     * Prepares the coins <span> for output
+     *
+     * @access    protected
+     * @param    array        $metadata: The metadata array
+     *
+     * @return    string        The coins <span> ready for output
+     */
+     protected function generateCoins(array $metadata)
+    {
+
+        // The output follows the "Brief guide to Implementing OpenURL 1.0 Context Object for Journal Articles"
+        // -> https://archive.is/a0Hgs
+        //
+        // TODO: Get the document type to differentiate info:ofi/fmt:kev:mtx:[book/journal] and rft.genre=[…]
+
+        $coins = 'url_ver=Z39.88-2004&ctx_ver=Z39.88-2004&rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&rft.genre=unknown';
```

Comment:
This line is very, very obscure. Can you split up the concatenation or find other ways to explain why this accumulator string has to start this way? Are the comments above actually related?

Reply:
Yes, the comment relates to this. This line is how the COinS-string has to start that by definition.

`ctx_ver=Z39.88-2004`
`rft_val_fmt=info:ofi/fmt:kev:mtx:journal`

and then all `key=values` follow.

Since I cannot determine the media type the first key=value in the form of `rft.genre=unknown`. That's where the line comes from. I can split that up to make it clearer. :)